### ISSUE TITLE
[Tracking] Post Spec Freeze Checklist 

### ISSUE THREAD

**Issue by nisdas** [prysmaticlabs/prysm#2943]
With all the important remianing items having been merged in to `spec-v0.6` branch, it is finally ready to be reviewed and merged. This issue will track whatever tasks that we will need to perform:

- [x] Runtime Fixes. With the large velocity of changes that have occured over the past few months our runtime is broken. The goal is to get runtime working again and to have a beacon node run for at least a 100 slots from genesis.
- [x] Mock out Genesis Trigger. With the genesis event being triggered by the node instead of the
chainstart log from the deposit contract. We can mock it out during runtime for easier startup for testing.
- [x] Clean Up unused/redundant tests, methods. This would involve looking through all the changes in the spec-v0.6 branch and compare that to our current codebase for code that is 
outdated.
- [x] Improve code coverage across the repo, especially in the core, blockchain, sync package. Since these are critical to the functioning of our node, we should be testing all possible code 
paths.
- [x] Begin refactor of db package, once this `spec-v0.6` is merged into master. We can start with
cleaning up our db package and abstracting it better

**Comment by terencechain** [comment]
- [ ] Go through all the helpers and processing code and verify. they are optimized and simplified using the best of our abilities. With the goal of removing unnecessary cache as it was a temporary patched solution

**Comment by prestonvanloon** [comment]
The mock for genesis trigger is done from the ETH1 partial mocked backend. 
cc: @zscole

**Comment by rauljordan** [comment]
This is a bit too open ended of an issue to be resolved - let's come up with smaller, specific tracking issues for the things outlined above.
